src/convnets/inception.jl

```diff
@@ -176,21 +178,392 @@ See also [`inception3`](#).
+                    conv_bn((1, 7), 256, 256; pad = (0, 3))...,
+                    conv_bn((7, 1), 256, 320; pad = (3, 0))...,
+                    conv_bn((3, 3), 320, 320; stride = 2)...)
+    branch3 = MaxPool((3, 3); stride = 2)
+    return Parallel(cat_channels, branch1, branch2, branch3)
+end
+
+function inceptionv4_c()
+    branch1 = Chain(conv_bn((1, 1), 1536, 256)...)
+    branch2 = Chain(conv_bn((1, 1), 1536, 384)...,
+                    Parallel(cat_channels,
+                             Chain(conv_bn((1, 3), 384, 256; pad = (0, 1))...),
+                             Chain(conv_bn((3, 1), 384, 256; pad = (1, 0))...)))
+    branch3 = Chain(conv_bn((1, 1), 1536, 384)...,
+                    conv_bn((3, 1), 384, 448; pad = (1, 0))...,
+                    conv_bn((1, 3), 448, 512; pad = (0, 1))...,
+                    Parallel(cat_channels,
+                             Chain(conv_bn((1, 3), 512, 256; pad = (0, 1))...),
+                             Chain(conv_bn((3, 1), 512, 256; pad = (1, 0))...)))
+    branch4 = Chain(MeanPool((3, 3); stride = 1, pad = 1), conv_bn((1, 1), 1536, 256)...)
+    return Parallel(cat_channels, branch1, branch2, branch3, branch4)
+end
+
+"""
+    inceptionv4(; inchannels = 3, dropout = 0.0, nclasses = 1000)
+
+Create an Inceptionv4 model.
+([reference](https://arxiv.org/abs/1602.07261))
+
+# Arguments
+
+  - inchannels: number of input channels.
+  - dropout: rate of dropout in classifier head.
+  - nclasses: the number of output classes.
+"""
+function inceptionv4(; inchannels = 3, dropout = 0.0, nclasses = 1000)
+    body = Chain(conv_bn((3, 3), inchannels, 32; stride = 2)...,
+                 conv_bn((3, 3), 32, 32)...,
+                 conv_bn((3, 3), 32, 64; pad = 1)...,
+                 mixed_3a(),
+                 mixed_4a(),
+                 mixed_5a(),
+                 inceptionv4_a(),
+                 inceptionv4_a(),
+                 inceptionv4_a(),
+                 inceptionv4_a(),
+                 reduction_a(),  # mixed_6a
+                 inceptionv4_b(),
+                 inceptionv4_b(),
+                 inceptionv4_b(),
+                 inceptionv4_b(),
+                 inceptionv4_b(),
+                 inceptionv4_b(),
+                 inceptionv4_b(),
+                 reduction_b(),  # mixed_7a
+                 inceptionv4_c(),
+                 inceptionv4_c(),
+                 inceptionv4_c())
+    head = Chain(GlobalMeanPool(), MLUtils.flatten, Dropout(dropout), Dense(1536, nclasses))
+    return Chain(body, head)
+end
+
+"""
+    Inceptionv4(; inchannels = 3, dropout = 0.0, nclasses = 1000)
+
+Creates an Inceptionv4 model.
+([reference](https://arxiv.org/abs/1602.07261))
+
+# Arguments
+
+  - inchannels: number of input channels.
+  - dropout: rate of dropout in classifier head.
+  - nclasses: the number of output classes.
+"""
+struct Inceptionv4
+    layers::Any
+end
+
+function Inceptionv4(; inchannels = 3, dropout = 0.0, nclasses = 1000)
```

Comment:
This should probably take the `pretrain` keyword for consistency.

Reply:
Will do for the `Inception` line of models, although right now the API for the newer models (like `ViT` or `ConvNeXt`) does not in fact, expose a pretrain option - will have to check that later depending on availability

Reply:
Hmm those probably should to but we can fix that later. The `loadpretrain!` mechanism already does the error handling for you. So you can unconditionally do the loading when `pretrain` is true.